### ISSUE TITLE
Combining control flow operators

### ISSUE THREAD

**Issue by VSjolund** [ryansolid/solid#41]
Does Solid support combining control flow operators? I can't get it to work, not sure what I'm doing wrong. I removed code bit by bit and I think this example captures where the error occurs:
```
import { createState, createRoot, onCleanup } from 'solid-js'

const CountingComponent = () => {
  const [state, setState] = createState({ counter: 0, data: [1, 2, 3], flag: 3 });

  const interval = setInterval(() => {
    setState({ counter: state.counter + 1, flag: state.counter % 3 })
  }, 1000);

  onCleanup(() => clearInterval(interval));

  return (
    <$ each={state.data} >
      <>
        <$ when={state.flag === 1}>
          <div>
            Flag is 1.
          </div>
        </$>
        <$ when={state.flag === 2}>
          <span>
            Flag is 2.
          </span>
        </$>
      </>
    </$>
  )
}

createRoot(() =>
  document
    .getElementById("app")
    .appendChild(<CountingComponent />)
);
```
Once data has been rendered once, it throws the same error 3 times (cannot read property nextSibling of null, in removeNodes), then goes on to get the next error (cannot read property __rGroup of null), where it enters a steady state and keeps throwing this error.

The error is thrown when trying to dismount the last item from `state.data`. The childnodes of `.app` are then `[#comment, div, #comment]`, and it passes in the last comment to removeNodes and null as the end marker. Looks like it's either sending in a remaining comment node that should have been removed from a previous item, or passing in the wrong comment node? Not sure.

**Comment by ryansolid** [comment]
Thank you for the detailed bug report. This is an area that I've been tinkering with the last few releases (reducing number of comment nodes). I probably introduced something, some newer hole in the logic. I will try to get this fixed as soon as I can, probably later today.

**Comment by ryansolid** [comment]
Alright, there was a bug around clear code getting the wrong markers. Fixed v0.8.4. See it working here: https://codesandbox.io/s/combining-flow-xnt7m
